lumberjack.go

```diff
@@ -324,9 +339,30 @@ func deleteAll(dir string, files []logInfo) {
 	}
 }
 
+// compressLogs compresses any uncompressed logs during the cleanup process
+func (l *Logger) compressLogs() {
+	l.cmu.Lock()
+	defer l.cmu.Unlock()
+	files, err := l.oldLogFiles(false)
+	if err != nil {
+		fmt.Errorf("Unable to compress log files: %s", err)
```

Comment:
This doesn't actually do anything.  It returns a new error that is then discarded.  Ideas on how to handle errors from here are welcome, for sure.   Generally one would just log them, but.... we're kind low level for that.  Maybe let the user pass in a method to handle compression failures?  I don't know.